src/term/mod.rs

```diff
@@ -941,40 +941,31 @@ impl Term {
         use std::ops::Range;
 
         trait Append : PushChar {
-            fn append(&mut self, grid: &Grid<Cell>, line: usize, cols: Range<Column>) -> Option<Range<Column>>;
+            fn append(&mut self, grid: &Grid<Cell>, line: usize, cols: Range<Column>);
         }
 
         impl Append for String {
-            fn append(
-                &mut self,
-                grid: &Grid<Cell>,
-                mut line: usize,
-                cols: Range<Column>
-            ) -> Option<Range<Column>> {
+            fn append(&mut self, grid: &Grid<Cell>, mut line: usize, cols: Range<Column>) {
                 // Select until last line still within the buffer
                 line = min(line, grid.len() - 1);
 
                 let grid_line = &grid[line];
                 let line_length = grid_line.line_length();
                 let line_end = min(line_length, cols.end + 1);
 
-                if cols.start >= line_end {
-                    None
-                } else {
+                if line_end.0 == 0 && cols.end >= grid.num_cols() - 1 {
+                    self.push('\n');
+                } else if cols.start < line_end {
                     for cell in &grid_line[cols.start..line_end] {
                         if !cell.flags.contains(cell::Flags::WIDE_CHAR_SPACER) {
                             self.push(cell.c);
                         }
                     }
 
-                    let range = Some(cols.start..line_end);
+                    let range = cols.start..line_end;
                     if cols.end >= grid.num_cols() - 1 {
-                        if let Some(ref range) = range {
-                            self.maybe_newline(grid, line, range.end);
-                        }
+                        self.maybe_newline(grid, line, range.end);
```

Comment:
Could this `range.end` just be `line_end`? Is `range` here even needed?

Reply:
Thanks for the review, it is not needed. I've already removed some old redundant code, but it seems like this was left over.